src/dyn/Energy_and_Forces.cpp

```diff
@@ -326,7 +361,41 @@ void update_forces(dyn_control_params& prms, dyn_variables& dyn_vars, nHamiltoni
 */
     }     
   }// Ehrenfest
+  else if(prms.force_method==3){  // QTSH forces
+    // Effective force determined by the TSH procedure 
+    vector<int> effective_states(dyn_vars.act_states);
+  
+    if(prms.enforce_state_following==1){ // NBRA-like enforcement: adiabatic dynamics, in terms of forces 
+      for(int itraj=0; itraj<ntraj; itraj++){ effective_states[itraj] = prms.enforced_state_index; }
+    }
+
+    // Diabatic 
+    if(prms.rep_force==0){  *dyn_vars.f = ham.forces_dia(effective_states).real();   }
+
+    // Adiabatic 
+    else if(prms.rep_force==1){  
+      *dyn_vars.f = ham.forces_adi(effective_states).real(); 
+    }// rep_force == 1
+
+    // Non-classical term calculations

```

Comment:
the term "non-classical" is confusing here - do you mean the total QTSH force, or only the Ehrenfest-like correction to the adiabatic forces? I think we could just use the regular forces variable to keep the entire QTSH force, if this method is used.